Accountant

BobTheBuidler opened the pull request.

mariuspod commented on yearn/treasury/accountant/cost_of_revenue/general.py:
```diff
@@ -0,0 +1,55 @@
+    hashes = {
+        Network.Mainnet: [
+            # Thought we automated these... why aren't they sorting successfully? 
+            ["0x590b0cc67ba42dbc046b8cbfe2d314fbe8da82f11649ef21cdacc61bc9752d83", IterFilter('log_index',[275,276,278])],
+            ["0xd1b925ad7fdd9abdd31460a346d081d6afe9f6cb1c1b0cd5f6129885edf318da", IterFilter('log_index',[174,177])],
+            ["0xe11b4e3ece520c1818ffe821c038779f87c293aa32c26115265b6b8fb23c30bd", Filter('log_index', 154)],
```
what are these magic numbers here ?

BobTheBuidler replied:
They are the log indexes of the particular value change components of each tx that need to be sorted

mariuspod reviewed (changes requested):
I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.

The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.

BobTheBuidler commented:
> I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.
> 
> The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.

ah yes, that was just supposed to be a local change, will revert

BobTheBuidler commented on yearn/__init__.py:
```diff
@@ -4,6 +4,11 @@
 setup_logging()
 setup_sentry()
-if network.is_connected():
+# hacky workaround when node doesn't support web3_clientVersion
+try:
+    if network.is_connected():
+        from yearn.middleware.middleware import setup_middleware
+        setup_middleware()
+except AssertionError:
```
Our nodes, or nodes generally? 
I don't recall offhand which one, but this was a fix for a node from one of the saas providers

mariuspod replied:
I checked the code again, `web3.clientVersion` is referenced in here in the sentry init code: https://github.com/yearn/yearn-exporter/blob/3a4ab9873dee063cb38e30cb387e0af0ad2343ca/yearn/sentry.py#L18

So maybe the `try..except` must be expanded as `setup_sentry` is called before it ?
You can also try to set the the clientVersion to some value if it's not available or create a check that it's not sent to sentry if n/a.

BobTheBuidler commented on services/dashboard/docker-compose.yml:
```diff
@@ -629,7 +700,7 @@ services:
   postgres:
     image: postgres
     ports:
-      - 127.0.0.1:5432:5432
+      - 5432:5432
```
yep that was an accident, will revert

BobTheBuidler replied:
reverted, resolving

BobTheBuidler commented on scripts/treasury_transactions_exporter.py:
```diff
@@ -0,0 +1,307 @@
+
+logger = logging.getLogger('yearn.treasury_transactions_exporter')
+
+BATCH_SIZE = {
+    Network.Mainnet: 25_000,
+    Network.Fantom: 2_000_000,
+    Network.Gnosis: 2_000_000,
+}[chain.id]
```
Ah yes, arb was added after this PR was started. Will add.

BobTheBuidler commented:
> > I've added some comments to some of your changes mainly targeting custom workarounds or magic numbers which are not critical.
> > The only critical thing is the exposure of the db port to the world inside docker-compose which I'd rather not do.
> 
> ah yes, that was just supposed to be a local change, will revert

reverted and pushed, marking this as resolved.

BobTheBuidler commented on Makefile:
```diff
@@ -3,11 +3,46 @@ ifdef FLAGS
-tvl_command := docker-compose --file services/tvl/docker-compose.yml --project-directory .
-test_command := docker-compose --file services/dashboard/docker-compose.test.yml --project-directory .
-all_command := docker-compose --file services/dashboard/docker-compose.yml --project-directory .
+# docker-compose commands
+dashboards_command 	:= docker-compose --file services/dashboard/docker-compose.yml --project-directory .
+tvl_command 		:= docker-compose --file services/tvl/docker-compose.yml --project-directory .
+test_command 		:= docker-compose --file services/dashboard/docker-compose.test.yml --project-directory .
+
+# treasury exporter convenience vars
+ethereum_treasury_containers := treasury-exporter 			historical-treasury-exporter 			treasury-transactions-exporter
+fantom_treasury_containers   := ftm-treasury-exporter 		historical-ftm-treasury-exporter		ftm-treasury-transactions-exporter
+arbitrum_treasury_containers := arbi-treasury-exporter 		historical-arbi-treasury-exporter		arbi-treasury-transactions-exporter
+gnosis_treasury_containers 	 := gnosis-treasury-exporter 	historical-gnosis-treasury-exporter		gnosis-treasury-transactions-exporter
+
+# less spammy output convenience vars
+ethereum_containers_lite := eth-exporter 	historical-eth-exporter 	 		sms-exporter 		historical-sms-exporter 		transactions-exporter 		$(ethereum_treasury_containers)
+fantom_containers_lite   := ftm-exporter 	historical-ftm-exporter 	 	ftm-sms-exporter 	historical-ftm-sms-exporter 	ftm-transactions-exporter 		$(fantom_treasury_containers)
+arbitrum_containers_lite := arbi-exporter 	historical-arbi-exporter 	 	arbi-sms-exporter 	historical-arbi-sms-exporter 	arbi-transactions-exporter 		$(arbitrum_treasury_containers)
+gnosis_containers_lite   := gnosis-exporter historical-gnosis-exporter 	 gnosis-sms-exporter historical-gnosis-sms-exporter 	gnosis-transactions-exporter 	$(gnosis_treasury_containers)
+
+# container vars
+ethereum_containers := $(ethereum_containers_lite) 	wallet-exporter 		partners-exporter
+fantom_containers 	:= $(fantom_containers_lite) 	ftm-wallet-exporter 	ftm-partners-exporter
+arbitrum_containers := $(arbitrum_containers_lite) 	arbi-wallet-exporter	arbi-partners-exporter
+gnosis_containers 	:= $(gnosis_containers_lite) 	gnosis-wallet-exporter	gnosis-partners-exporter
+
+treasury_containers := $(ethereum_treasury_containers) $(fantom_treasury_containers) $(arbitrum_treasury_containers) $(gnosis_treasury_containers)
+all_containers := $(ethereum_containers) $(fantom_containers) $(arbitrum_containers) $(gnosis_containers)
+
+# Basic Makefile commands
+dashboards: dashboards-up
+tvl: tvl-up
+
+logs:
+	$(dashboards_command) logs -f -t $(ethereum_containers_lite) $(fantom_containers_lite) $(arbitrum_containers_lite) $(gnosis_containers_lite)
```
Resolving per discussions with @mariuspod  in DM's
